### DISCUSSION TITLE
Issue with "Sidebar Layout - Narrow Sidebar with Titles"

### DISCUSSION THREAD

**Discussion by marcosgeorgopoulos** [primefaces/community#384] [PrimeReact Blocks] [answered]
Hi,

I've purchased prime blocks and I'm trying to use one of the sidebar layout examples and dynamically update the content in the main 'dashed' component area. That all works fine and the relavent section of code looks like

 ```
 <div className="p-5 flex flex-column flex-auto">
                <div className="border-2 border-dashed surface-border border-round surface-section flex-auto">
                   <main>{childpage}</main> 
                </div>    
 </div>
```

However what is happening now is the main content area is going down past the height of the sidebar. 

<img width="1134" alt="Screenshot 2023-11-05 at 7 08 46 pm" src="https://github.com/primefaces/community/assets/5653675/f20a5049-6493-4880-8d5c-6d5479b427e9">

What I'm wanting to accomplish is to have the content to not resize greater then the original size and have a vertical scrolling bar if the content is too large.

Here it is without content.

<img width="1134" alt="Screenshot 2023-11-05 at 7 21 39 pm" src="https://github.com/primefaces/community/assets/5653675/21f3f7e8-1d49-49bf-8130-a07c2a830bec">

I appreciate this is probably a very basic question, but I'm new to prime react, prime flex etc and was hoping for some assistance.

cheers.

**Comment by melloware** [comment]
Can you put together a StackBlitz showingthe issue: https://stackblitz.com/edit/react-ozgtgi?file=src%2FApp.js

**Reply by marcosgeorgopoulos** [reply]
Thank you I've managed to simulate the issue here https://stackblitz.com/edit/react-ozgtgi-bysxev?file=src%2FApp.jsx

I use 'open in new tab'.  I've duplicated a bunch of edit boxes etc to make the issue more obvious. I seem to be missing some styling on the form components in this example, but I don't think that is relevant. 

It's rendering the layout component ( component->Layout.jsx ) which then renders pages->Profile.jsx

I'm trying to get the main page content to not extend past the menu at the bottom.

I've tried using overflow-auto and overflow-scroll in <main> and in Profile.jsx but its not working.

I am new to both React and this library so I suspect its something simple I'm missing.

thank you.

**Comment by SelenKockar** [comment, marked as answer]
Hello, 
After flex-auto you can give maximum height with  max-h-...  as you like after that you can add overflow-y-auto class for scroll bar. 

               <div className="border-2 border-dashed surface-border border-round surface-section flex-auto">
                  <main>{childpage}</main> 
               </div>

**Reply by marcosgeorgopoulos** [reply]
Thank you, I did try that but could not get it working. I have uploaded an example of what I'm seeing to the link in the other reply.

cheers.

**Reply by nitrogenous** [reply]
We are truly sorry about the delayed answer. I simply achieved the thing you were mentioning by just deleting the 'h-screen' class. I hope this will help.

![image](https://github.com/primefaces/community/assets/29578451/11ec2abf-7c30-40c7-b9a2-6795f5b7b76f)
